qiskit_experiments/library/calibration/remsey_xy.py

```diff
@@ -0,0 +1,210 @@
+                Rz rotation. This quantity is given in Hz.
+        """
+        super().__init__([qubit])
+
+        delays = delays or self.experiment_options.delays
+        self.set_experiment_options(delays=delays, unit=unit, osc_freq=osc_freq)
+
+    def _pre_circuit(self) -> QuantumCircuit:
+        """Return a preparation circuit.
+
+        This method can be overridden by subclasses e.g. to calibrate schedules on
+        transitions other than the 0 <-> 1 transition.
+        """
+        return QuantumCircuit(1)
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the Ramsey XY calibration experiment.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            A list of circuits with a variable delay.
+
+        Raises:
+            AttributeError: if unit is 'dt', but 'dt' the parameter is missing
+                from the backend's configuration.
+        """
+
+        conversion_factor = 1
+        if self.experiment_options.unit == "dt":
+            try:
+                conversion_factor = getattr(backend.configuration(), "dt")
+            except AttributeError as no_dt:
+                raise AttributeError(
+                    "Dt parameter is missing from the backend's configuration."
+                ) from no_dt
+
+        elif self.experiment_options.unit != "s":
+            conversion_factor = apply_prefix(1, self.experiment_options.unit)
+
+        # Compute the rz rotation angle to add a modulation.
+        p_delay = Parameter("delay")
+
+        rotation_angle = 2 * np.pi * self.experiment_options.osc_freq * conversion_factor * p_delay
+
+        # Create the X and Y circuits.
+        ram_x = self._pre_circuit()
+        ram_x.sx(0)
+        ram_x.delay(p_delay, 0, self.experiment_options.unit)
+        ram_x.rz(rotation_angle, 0)
+        ram_x.sx(0)
+        ram_x.measure_active()
+
+        ram_y = self._pre_circuit()
+        ram_y.sx(0)
+        ram_y.delay(p_delay, 0, self.experiment_options.unit)
+        ram_y.rz(rotation_angle - np.pi / 2, 0)
+        ram_y.sx(0)
+        ram_y.measure_active()
+
+        # Add the schedule if any.
+        schedule = self.experiment_options.schedule
+        if schedule is not None:
+            for circ in [ram_x, ram_y]:
+                circ.add_calibration("sx", self.physical_qubits, schedule)
+
+        circs = []
+        for delay in self.experiment_options.delays:
+
+            metadata = {
+                "experiment_type": self._type,
+                "qubits": self.physical_qubits,
+                "osc_freq": self.experiment_options.osc_freq,
+                "unit": "s",
+                "xval": delay * conversion_factor,
+            }
+
+            # create ramsey x
+            assigned_x = ram_x.assign_parameters({p_delay: delay}, inplace=False)
+            assigned_x.metadata = metadata.copy()
```

Comment:
When you assign parameters with `inplace=False` the circuit copy method is called and the metadata dict is deepcopied. So you can avoid this copy.
https://github.com/Qiskit/qiskit-terra/blob/f87baa692765886da45becdc5f0fa5fff0429ab5/qiskit/circuit/quantumcircuit.py#L2026-L2027

Reply:
Thanks, done here: cfc849e